File: Neetcode 150/Two Pointer/valid-palindrome.cpp

// https://leetcode.com/problems/valid-palindrome/

class Solution {
public:

    bool checkPalindrome(string s){
        int size = s.size();
        int i = 0, j = size-1;
        while(i < j){
            if(s[i++] != s[j--]) return false;
        }

        return true;
    }

    bool isPalindrome(string str) {
        
        string cleanedString = "";
        for(auto  s : str){
            if(isalnum(s)) cleanedString += tolower(s);
        }

        // cout<<cleanedString;

        return checkPalindrome(cleanedString);
    }
};

File: Neetcode-150-js/Two Pointers/valid-palindrome.js

// https://leetcode.com/problems/valid-palindrome/description/

/**
 * @param {string} s
 * @return {boolean}
 */
var getAlNum = function(s){
  let ans = "";
  for(let i = 0; i < s.length; i++){
    if(s[i].match(/[a-z0-9]/)){
      ans += s[i];
    }
  }
  return ans;
}

var checkPalindrome = function(s){
  let i = 0, j = s.length-1;
  while(i <= j){
    if(s[i++] !== s[j--]) return false;
  }
  return true;
}

var isPalindrome = function(s) {
    const sLowerTrim = s.toLowerCase().trim();
    const sAlNum = getAlNum(sLowerTrim);
    return checkPalindrome(sAlNum);
};